app/views/layouts/application.html.erb

```diff
@@ -1,5 +1,7 @@
 <%= render "shell/top" %>
-<div id="base-payment-pointer" data-payment-pointer="<%= SiteConfig.payment_pointer %>"></div>
+<% if SiteConfig.payment_pointer.present? %>
```

Comment:
We have a lot of conditional checks for these SiteConfigs scattered all over the code. I'm wondering if it might be cleaner to hook up a `method_missing` method to handle and return something when a SiteConfig is not present.

Reply:
Many of these conditionals are in views and I'm not sure `method_missing` can help us with that, as returning view snippets from there does not seem particularly clean either.

I thought along the lines of doing a refactor later where we extract the conditional code into partials, so the logic on whether or not to render something is self-contained and the including view is conditional free. For this specific example:

```ruby
<%= render partial: "payment_pointer", locals: { payment_pointer: SiteConfig.payment_pointer } %>
```

The partial then can be something like

```ruby
<% if payment_pointer.present? %>
  ...
<% end %>
```

That's a little bit like frontend components, i.e. the enclosing "component" just renders and something and passes in data (here locals instead of props) and the "component" itself decides if/what to render.

How does that sound?